README.md

```diff
@@ -131,6 +131,68 @@ func main() {
 }
 ```
 
+## Provide a custom logger
+
+If you want to provide a logging implementation to the SDK to capture errors and other application logs, you can do so by passing in an implementation of the `ApplicationLogger` interface to the `InitClient` function.
+
+You can use the `eppoclient.ScrubbingLogger` to scrub PII from the logs.
+
+```go
+import (
+    "github.com/Eppo-exp/golang-sdk/v4/eppoclient"
+    "github.com/sirupsen/logrus"
+)
+
+type LogrusApplicationLogger struct {
+    logger *logrus.Logger
+    logLevel logrus.Level
+}
+
+func NewLogrusApplicationLogger(logLevel logrus.Level) *LogrusApplicationLogger {
+    logger := logrus.New()
+    logger.SetFormatter(&logrus.JSONFormatter{})
+    return &LogrusApplicationLogger{logger: logger, logLevel: logLevel}
+}
+
+func (l *LogrusApplicationLogger) Debug(args ...interface{}) {
+    if l.logLevel <= logrus.DebugLevel {
+        l.logger.Debug(args...)
+    }
+}
+
+func (l *LogrusApplicationLogger) Info(args ...interface{}) {
+    if l.logLevel <= logrus.InfoLevel {
+        l.logger.Info(args...)
+    }
+}
+
+func (l *LogrusApplicationLogger) Warn(args ...interface{}) {
+    if l.logLevel <= logrus.WarnLevel {
+        l.logger.Warn(args...)
+    }
+}
+
+func (l *LogrusApplicationLogger) Error(args ...interface{}) {
+    if l.logLevel <= logrus.ErrorLevel {
+        l.logger.Error(args...)
+    }
+}
+
+func main() {
+  // Initialize a custom logger; example using logrus
+  // Set log level to Info
+  applicationLogger := NewLogrusApplicationLogger(logrus.InfoLevel)
+  scrubbingLogger := eppoclient.NewScrubbingLogger(applicationLogger)
```

Comment:
@guppykang I decided to provide this `ScrubbingLogger` for you to use optionally - other customers (or you if debugging) may want to show the `SdkKey` to console but I can appreciate that in production it could be desirable to mask it.